easysocial/easysocial/pages.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage  Com_api-plugins
+ *
+ * @copyright   Copyright (C) 2009-2014 Techjoomla, Tekdi Technologies Pvt. Ltd. All rights reserved.
+ * @license     GNU GPLv2 <http://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
+ * @link        http://techjoomla.com
+ * Work derived from the original RESTful API by Techjoomla (https://github.com/techjoomla/Joomla-REST-API)
+ * and the com_api extension by Brian Edgerton (http://www.edgewebworks.com)
+ */
+defined('_JEXEC') or die('Restricted access');
+
+jimport('joomla.plugin.plugin');
+jimport('joomla.html.html');
+
+require_once JPATH_SITE . '/plugins/api/easysocial/libraries/mappingHelper.php';
+
+/**
+ * API class EasysocialApiResourcePages
+ *
+ * @since  1.0
+ */
+class EasysocialApiResourcePages extends ApiResource
+{
+	/**
+	 * Method to get pages list
+	 *
+	 * @return  mixed
+	 *
+	 * @deprecated 2.0 use post instead
+	 *
+	 * @since 1.0
+	 */
+	public function get()
+	{
+		$this->post();
+	}
+
+	/**
+	 * Method to get pages list
+	 *
+	 * @return  JResponse object
+	 *
+	 * @since 2.0
+	 */
+	public function post()
+	{
+		$app = JFactory::getApplication();
+		$input = $app->input;
+		$filters = $input->get("filters", array(), "ARRAY");
+		$user = ES::user();
+
+		$res = new stdclass;
+		$res->result = array();
+		$res->empty_message = '';
+
+		$limit		=	$app->input->get('limit', 10, 'INT');
+		$limitstart	=	$app->input->get('limitstart', 0, 'INT');
+		$filters['limit'] = $limit;
+
+		// Set default filters
+		$filters['state']	= isset($filters['state']) ? $filters['state'] : SOCIAL_CLUSTER_PUBLISHED;
+		$filters['types']	= isset($filters['types']) ? $filters['types'] : $user->isSiteAdmin() ? 'all' : 'user';
+		$filters['ordering']	= isset($filters['ordering']) ? $filters['ordering'] : 'latest';
+
+		$model = ES::model('Pages');
+		$MappingHelper = new EasySocialApiMappingHelper;
+		$pages = array();
+
+		$pages = $model->getPages($filters);
+		$pages = $MappingHelper->mapItem($pages, 'page', Jfactory::getUser()->id);
+
+		if ($limit)
+		{
+			$pages		=	array_slice($pages, $limitstart, $limit);
+		}
+
+		if (empty($pages))
+		{
+			if (! empty($filters['all']))
```

Comment:
@Pratikshakhandagale We can use switch case here instead of long list of if else

Reply:
@snehal0904 - No, Here I have checked only boolean values of filters.

Reply:
@snehal0904 - No, Here I have checked only boolean values of filters.